packages/experimental/Divider/src/Divider.styling.ts

```diff
@@ -0,0 +1,163 @@
+import { useMemo } from 'react';
+import type { ViewProps, ColorValue } from 'react-native';
+import type { Theme } from '@fluentui-react-native/framework';
+import type { IconPropsV1 as IconProps } from '@fluentui-react-native/icon';
+import type { TextProps } from '@fluentui-react-native/text';
+import type { DividerTokens, DividerProps } from './Divider.types';
+
+const getIconProps = (contentColor: ColorValue, icon: IconProps): IconProps => {
+  if (icon.fontSource) {
+    return {
+      fontSource: {
+        ...icon.fontSource,
+        color: contentColor,
+      },
+    };
+  } else if (icon.svgSource) {
+    return {
+      svgSource: {
+        ...icon.svgSource,
+        color: contentColor,
+      },
+    };
+  } else {
+    throw new Error('IconProps require either a fontSource or svgSource; neither has been passed.');
+  }
+};
+
+export const useDividerSlotProps = (props: DividerProps, tokens: DividerTokens) => {
+  const rootProps: ViewProps = useMemo(
+    () => ({
+      style: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        display: 'flex',
+        minWidth: tokens.minWidth,
+        maxWidth: tokens.maxWidth,
+        minHeight: tokens.minHeight,
+        maxHeight: tokens.maxHeight,
+        padding: tokens.padding,
+        paddingStart: tokens.paddingStart,
+        paddingEnd: tokens.paddingEnd,
+        paddingHorizontal: tokens.paddingHorizontal,
+        paddingVertical: tokens.paddingVertical,
+        ...(props.vertical
+          ? {
+              flexDirection: 'column',
+              paddingVertical: props.insetSize,
+              height: '100%',
+            }
+          : {
+              flexDirection: 'row',
+              paddingHorizontal: props.insetSize,
+            }),
+      },
+    }),
+    [
+      props.vertical,
+      props.insetSize,
+      tokens.minHeight,
+      tokens.maxHeight,
+      tokens.minWidth,
+      tokens.maxWidth,
+      tokens.padding,
+      tokens.paddingHorizontal,
+      tokens.paddingVertical,
+      tokens.paddingStart,
+      tokens.paddingEnd,
+    ],
+  );
+
+  const beforeLineProps: ViewProps = useMemo(
+    () => ({
+      style: {
+        flexBasis: tokens.minLineSize,
+        flex: tokens.flexBefore,
+        borderColor: tokens.lineColor,
+        borderStyle: 'solid',
+        ...(props.vertical
+          ? { borderLeftWidth: tokens.thickness, minHeight: tokens.minLineSize }
+          : { borderTopWidth: tokens.thickness, minWidth: tokens.minLineSize }),
+      },
+    }),
+    [tokens.flexBefore, tokens.lineColor, tokens.minLineSize, tokens.thickness, props.vertical],
+  );
+
+  const afterLineProps: ViewProps = useMemo(
```

Comment:
verification with RTL would be great, i believe every would work fine 😄

Reply:
How can I verify RTL behavior? Currently for text, it remains centered in its container for non RTL languages.

Reply:
Hmm interesting, Actually that's would be task on Mobile Platform as they have insetSpacing on one side. 
I will verify once i make change for mobile.